conformance_tests/sysman/test_sysman_frequency/src/test_sysman_frequency.cpp

```diff
@@ -408,6 +412,48 @@ TEST_F(
     }
   }
 }
+
+TEST_F(
+    FREQUENCY_TEST,
+    GivenValidFrequencyHandleWhenRequestingSetFrequencyWithInvalidRangeThenExpectMinAndMaxFrequencyAreClampedToHardwareLimits) {
+  for (auto device : devices) {
+    uint32_t p_count = 0;
+    auto pfreq_handles = lzt::get_freq_handles(device, p_count);
+    if (p_count == 0) {
+      FAIL() << "No handles found: "
+             << _ze_result_t(ZE_RESULT_ERROR_UNSUPPORTED_FEATURE);
+    }
+
+    for (auto pfreq_handle : pfreq_handles) {
+      EXPECT_NE(nullptr, pfreq_handle);
+
+      zes_freq_range_t invalid_freq_range = {};
+      zes_freq_range_t clamped_freq_range = {};
+
+      zes_freq_properties_t freq_property =
+          lzt::get_freq_properties(pfreq_handle);
+
+      if (freq_property.canControl) {
+        if (freq_property.min - 100 >= 0) {
+          invalid_freq_range.min = freq_property.min - 100;
+        } else {
+          invalid_freq_range.min = 0;
+        }
+        invalid_freq_range.max = freq_property.max + 100;
+
+        lzt::set_freq_range(pfreq_handle, invalid_freq_range);
+        clamped_freq_range = lzt::get_and_validate_freq_range(pfreq_handle);
+        EXPECT_EQ(freq_property.min, clamped_freq_range.min);
+        EXPECT_EQ(freq_property.max, clamped_freq_range.max);
+      } else {
+        LOG_WARNING << "User cannot control min/max frequency setting for "
+                       "frequency domain "
+                    << freq_property.type;
```

Comment:
Can this be printed as string rather than enumeration to be user friendly log ?

Reply:
Updated